### PR TITLE
Fixing AngryTock API calls

### PR DESCRIPTION
**Pull request by rogeruiz**
This work reflects the API changes introduced into Tock at 18F/tock#720 and 18F/tock#729.

### CODE REVIEW

**Comment by rogeruiz** [comment]
Getting the following error here.

```sh
   2018-03-13T15:34:09.19-0400 [APP/PROC/WEB/0] ERR 2018/03/13 19:34:09 This is the URL [https://tock.18f.gov/api/reporting_period_audit.json]
   2018-03-13T15:34:09.29-0400 [APP/PROC/WEB/0] ERR 2018/03/13 19:34:09 json: cannot unmarshal array into Go value of type tockPackage.ReportingPeriodAuditList
   2018-03-13T15:34:09.29-0400 [APP/PROC/WEB/0] ERR 2018/03/13 19:34:09 data&{{0  } []}
   2018-03-13T15:34:09.30-0400 [APP/PROC/WEB/0] ERR panic: runtime error: index out of range
```

**Comment by rogeruiz** [comment]
@geramirez any ideas on how to adjust the marshaling on this?

**Comment by rogeruiz** [comment]
Alrighty, this is now ready for a review @annalee @ccostino @toolness. API calls are working properly. Turns out I was pushing the data into the wrong thing. 😁

File: tock/tock.go [hunk 1]
```diff
@@ -21,9 +21,11 @@ type User struct {
...
 // ReportingPeriod is a struct representation of the reporting_period JSON object from tock
 type ReportingPeriod struct {
-	StartDate    string `json:"start_date"`
-	EndDate      string `json:"end_date"`
-	WorkingHours int    `json:"working_hours"`
+	StartDate         string `json:"start_date"`
+	EndDate           string `json:"end_date"`
+	ExactWorkingHours int    `json:"exact_working_hours"`
+	MinWorkingHours   int    `json:"min_working_hours"`
+	MaxWorkingHours   int    `json:"max_working_hours"`
```
Comment (rogeruiz) [inline tock/tock.go:28]: @toolness here’s the struct that conincides with the API.

File: tock/tock.go [hunk 2]
```diff
@@ -37,7 +39,7 @@ type APIPages struct {
 //the Reporting Period Audit list endpoint
 type ReportingPeriodAuditList struct {
 	APIPages
-	ReportingPeriods []ReportingPeriod `json:"results"`
+	ReportingPeriods []ReportingPeriod
```
Comment (rogeruiz) [inline tock/tock.go:42]: Here’s another one @toolness this used to return a JSON object with a results property. Though now it just returns an Array.

Comment (rogeruiz) [reply tock/tock.go:42]: Looks like line 49 also needs this update

**Comment by rogeruiz** [comment]
Have to bring back the `UserApplier` function in `tock.go` as it’s used ~~once~~ _in_ `bot.go`. 🤦‍♂️

File: tock/tock.go (hunk 1 above)
Comment (toolness) [reply tock/tock.go:28]: This seems reasonable to me!

**Review by toolness — APPROVED** [review]
This is fine.

Thanks!

**Pull request merged**